robotiq_c_model_control/src/robotiq_c_model_control/RobotiqServiceInterface.py

```diff
@@ -0,0 +1,305 @@
+import time
+import rospy
+import ready_logging
+from threading import (
+    Lock,
+    Thread
+)
+from copy import deepcopy
+from robotiq_c_model_control.msg import (
+    CModel_robot_output
+)
+from robotiq_c_model_control.srv import (
+    Open,
+    OpenResponse
+)
+from robotiq_c_model_control.baseCModel import BaseCModel
+
+
+fault_text = {'/0x05': 'Action delayed, activation (reactivation) must be completed prior to renewed action.',
+              '/0x07': 'The activation bit must be set prior to action.',
+              '/0x08': 'Maximum operating temperature exceeded, wait for cool-down.',
+              '/0x0A': 'Under minimum operating voltage.',
+              '/0x0B': 'Automatic release in progress.',
+              '/0x0C': 'Internal processor fault.',
+              '/0x0D': 'Activation fault, verify that no interference or other error occurred.',
+              '/0x0E': 'Overcurrent triggered.',
+              '/0x0F': 'Automatic release completed.',
+              }
+
+
+@ready_logging.logged
+class RobotiqServiceInterface(BaseCModel):
+    has_goal_ = False
```

Comment:
This class should have some kind of comments to explain it's function. It looks pretty stateful.